README.md

```diff
@@ -0,0 +1,387 @@
+
+In the Core Architecture description we applied all 4 pillars of the OOP of Encapsulation, Inheritance, Polymorphism & Abstraction for the project.
+
+The team encountered a major bottleneck later in the project lifecycle and the original scope needed to be reshaped. During a few team meetings, we decided to reduce the number of tasks and combine certain processes. This would enhance the project while also making more attainable to achieve the goals set out. 
+
+UML Diagrams were updated to account for this with the User -> TaskManager, ProjectManager & UserManager -> with the improved StateManager. This streamlined the structure of the project and was one of the most vital implementation decisions made.
+
+The bottlenecks were also removed with the team effort on additional reviewing & fixing, as well removing unnecessary tasks no longer needed with the improved structure.
+
+JSON is used to handle the data involved.
+
+![](https://webteic.info/images/PMScore-7-1.jpg)
+> Screenshot: Team development workload improvements.
+
+During team meetings, everyone had their voice to address any issues and the team began to gel as one unit to reach the goals set out. Each person showed their own leadership abilities, support & working as one to deliver the project.
+
+###  Testing
+
+Rigorous testing (unit tests) was done from the outset with each task & classes. While this did slow down the overall development, it was better this way to allow a cleaner codebase that was not riddled with bugs.
+
+The code base was enhanced with additional comments to layout a clear structure for each file & class.
+
+![](Docs/UnitTestStructure.png)
+> Screenshot: Unit Testing structure.
+
+Various issues were found during the tests and were corrected, including refinements along the way. When a pull request was sent, another team member would jump in to review this pull request as a rule until the task & testing was complete.
+
+Admin, Developer, ProjectManager, and Tester roles all had their unit tests applied, validating the permissions off the different user roles, the actions each user role can do and to ensure functionality works as expected.
+
+xUnit testing tool for the .NET Framework was used with the Arrange-Act-Assert pattern.
+
+### Unit Test Coverage
+![](Docs/UnitTestCoverage.png)
+> Screenshot: Unit Testing coverage.
+
+## Deployment Instructions
+
+The following outlines the instructions to deploy the program:
+
+1. **Cloning the Repository:**
+
+Open Visual Studio 2022 and navigate to View - Terminal.
+
+Provided you have a terminal such as PowerShell, Command Prompt or Git Bash, open it and run the following command:
+
+git clone https://github.com/DistroRickDev/PMS-Core.git
+cd PMS-Core
+
+2. **Restoring Dependencies:**
+
+Run the following command:
+
+dotnet restore
+
+3. **Building the Solution:**
+
+Use Build - Build Solution in Visual studio or this command:
+
+dotnet build
+
+4. **Running the project:**
+
```

Comment:
we might add that: first navigate to the PMSCore project folder

Reply:
it's in line 273